src/main/java/ch/uzh/ifi/hase/soprafs22/service/GameService.java

```diff
@@ -26,10 +27,123 @@
 public class GameService {
     private final Logger log = LoggerFactory.getLogger(GameService.class);
     private final UserRepository userRepository;
+    private final Map<Long, Game> gameMap = new HashMap<>();
```

Comment:
We are storing the game in the lobby at the moment, so this would never be filled
eg. lobbymanger > get lobby > get game

Reply:
Why are we storing games inside lobbies?

Reply:
I would say because there exists a one-to-one connection between lobbies and a game. But I am open for a gameManger or something like that.

Reply:
> Why are we storing games inside lobbies?

because we set all the relevant game settings in the lobby and start the game from the lobby.

Reply:
furthermore: gameMap collides with the name for the actual map of the game (where the units are placed and the game takes place etc pp) which is also named gameMap

Reply:
imo it makes more sense that the lobby creates a game with the settings:

1. Lobbies should be deleted once promoted into games.
2. It makes no sense to interact with a lobby once the game has started.
3. All the mutable settings in lobby should be immutable in the game.

Keeping them together opens lots of questions/checks to add in lobby.
- Is changing ones name in the lobby OK while the game is ongoing?
- Can I leave a lobby and not leave the game?
- What happens to the lobby if I set the game to ranked?

As for a gameManager, there is no need for it, gameService is fine.

You decide, I can add lobbyService to gameService and get it from there but then everything will go through lobby and we will end up with a lobby god class.

Reply:
Good thought. I'll think about this.